### PR TITLE
Sigma constants check

### PR DESCRIPTION
**Pull request by gostkin**

### CODE REVIEW

File: src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala [hunk 1]
```diff
@@ -107,6 +107,10 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
     validateStateful(boxesToSpend, dataBoxes, stateContext, accumulatedCost).result.toTry
   }
 
+  def getByteLength[T](b: T, serializer: ScorexSerializer[T]): Long = {
+    return serializer.toBytes(b).length
```
Comment (oskin1) [inline src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala]: `return` is redundant syntax

File: src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala [hunk 2]
```diff
@@ -120,6 +124,27 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
       .validate(txManyInputs, inputs.size <= Short.MaxValue, s"$id: ${inputs.size}")
       .validate(txManyDataInputs, dataInputs.size <= Short.MaxValue, s"$id: ${dataInputs.size}")
       .validate(txManyOutputs, outputCandidates.size <= Short.MaxValue, s"$id: ${outputCandidates.size}")
+      .validateSeq(outputCandidates)((validationState, out) => validationState
+        .validate(txMaxBoxSize, getByteLength[ErgoBox](out.toBox(id, 0), ErgoBoxSerializer) <= ErgoConstants.MaxBoxSize.get)
+        .validate(txAdditionalRegistersCount, out.additionalRegisters.size <= ErgoBox.nonMandatoryRegistersCount)
+        .validateSeq(out.additionalRegisters)((validationState, reg) => validationState
```
Comment (oskin1) [inline src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala]: Please break the line after arrow ascription (`=>`)

File: src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala [hunk 3]
```diff
@@ -120,6 +124,27 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
...
+          .validate(txRegisterData, reg._2 match {
+            case (coll: ConcreteCollection[_]) =>
+              val w = SigmaSerializer.startWriter()
+              ValueSerializer.serialize(coll, w)
+              w.length <= ErgoConstants.MaxByteArrayLength.get
```
Comment (oskin1) [inline src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala]: Please avoid using `.get` on `Option`s and `Try`s. Using it outside `Try` is unsafe.

File: src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala [hunk 4]
```diff
@@ -120,6 +124,27 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
...
+            case (constant: Constant[_]) =>
+              constant.tpe match {
+                case l: SBigInt.type =>
+                  val w = SigmaSerializer.startWriter()
+                  DataSerializer.serialize(constant.value, constant.tpe, w)
+                  w.length <= ErgoConstants.MaxBigIntSizeInBytes.get
```
Comment (oskin1) [inline src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala]: Same as above

File: src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala (hunk 3 above)
Comment (gostkin) [reply src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala]: It's not an Option or Try, it is SizeConstant class that always can run get

File: src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala (hunk 4 above)
Comment (gostkin) [reply src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala]: It's not an `Option` or `Try`, it is `SizeConstant` class that always can run `.get`

**Comment by catena2w** [comment]
This checks will remain to be done inside sigmastate-interpreter

**Pull request closed without merging**